### PR TITLE
Fix wiab automation

### PR DESCRIPTION
**Pull request by amitsagtani97**
https://wearezeta.atlassian.net/browse/WPB-11340

Fix the autodeploy script for installing wire-server on fresh vms

### CODE REVIEW

File: bin/autodeploy.sh [hunk 1]
```diff
@@ -134,12 +136,22 @@ system_cleanup() {
   rm -f /home/$DEMO_USER/wire-server-deploy-static-*.tgz
 }
 
+# Make sure the host machine is up to date and has all required packages installed
+ssh -p "$SSH_PORT" "$SSH_USER"@webapp."$TARGET_SYSTEM" "
+  sudo apt update && sudo apt upgrade -y && sudo apt install -y aptitude apt-transport-https bind9-host curl debian-goodies dnsutils git dnsmasq less lsof net-tools rsyslog screen sudo vim wget whois docker.io telnet python3-lxml qemu qemu-kvm qemu-utils libvirt-clients libvirt-daemon-system virtinst bridge-utils
```
Comment (jschumacher-wire) [inline bin/autodeploy.sh:141]: I think we should let the Hetzner ansible playbook take care this.

You want to make sure that the `system_cleanup()` function does not fail when trying to run the script on a clean host, right?

I would create an `if` branch within that function to determine if the commands are available. Something like this:

```
system_cleanup() {
  if which docker > /dev/null && which virsh > /dev/null
  then
    for VM in $(virsh list --all --name); do virsh destroy "$VM"; virsh undefine "$VM" --remove-all-storage; done
    docker system prune -a -f
  fi
  rm -f /home/$DEMO_USER/.ssh/known_hosts
  rm -rf /home/$DEMO_USER/wire-server-deploy
  rm -f /home/$DEMO_USER/wire-server-deploy-static-*.tgz
}
```

**Review by jschumacher-wire — CHANGES_REQUESTED** [review]
Hi Amit, thank you for the fixes! I left a small suggestion.

File: bin/autodeploy.sh (hunk 1 above)
Comment (amitsagtani97) [reply bin/autodeploy.sh]: @jschumacher-wire I have updated the script to handle more cases, PTAL

File: .github/workflows/deploy-wiab.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+name: Deploy on Hetnzer WIAB setup
```
Comment (jschumacher-wire) [inline .github/workflows/deploy-wiab.yml:1]:
```suggestion
name: Deploy on Hetzner WIAB setup
```

**Review by jschumacher-wire — CHANGES_REQUESTED** [review]

**Comment by jschumacher-wire** [comment]
I've added a small check in `bin/offline-vm-setup.sh` to wait and only exit the script once VM deployment is done. Otherwise, we might kill the small webserver used by cloud-init too early, potentially resulting in failed bootstrap installs for the last few VMs on the list.

Side note - I remember there was discussion regarding where to implement the pipeline, and that Concourse might be preferred. I realize it's not in the ticket, but I wanted to double-check.

@lwille what do you think?

In any case, very good work @amitsagtani97 :)

File: .github/workflows/deploy-wiab.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        id: get_commit_sha
+        run: |
+          COMMIT_SHA=$(git rev-parse HEAD)
+          echo "commit_sha=$COMMIT_SHA" >> $GITHUB_ENV
+
+      # Step 4: Run the autodeploy script
+      - name: Run Auto Deploy Script
+        run: |
+          cd bin
+          ./autodeploy.sh --artifact-hash 2e8e4319ff3a8c309dc0ef5a1800611faa1ab0cb --target-domain wiab-test-box.wire.link --force-redeploy
```
Comment (jschumacher-wire) [inline .github/workflows/deploy-wiab.yml:37]: Shouldn't the `artifact-hash` ID be invoked dynamically from `$COMMIT_SHA` above?

Comment (amitsagtani97) [reply .github/workflows/deploy-wiab.yml]: nice catch, thanks. I forgot to remove the hard coded sha used for testing. Fixed now

**Review by jschumacher-wire — APPROVED** [review]

**Pull request merged**